### DISCUSSION TITLE
Sampler attributes with Phoenix

### DISCUSSION THREAD

**Discussion by chad-g-adams** [open-telemetry/opentelemetry-erlang-contrib#93] [Q&A]
When using `opentelemetry_phoenix`, is there any way to control which attributes are sent to the sampler? In my case, I would like to inject a new attribute (calculated based on details from the headers of the HTTP request), and have this attribute available to my sampler as it will affect the sampling decision.

Having access to the `conn` in the sampler would be another way to do what I need...

**Comment by tsloughter** [comment]
@chad-g-adams there isn't but there should be (may not be in the Phoenix library yet) an option for telling the instrumentation library to include certain headers as attributes. Maybe that should also include an optional transformation function, which would allow you to calculate the new attribute and it would be included in the attributes passed to the sampler.

**Comment by bryannaegele** [comment]
You'll be able to use filters in an upcoming release which will act as simple predicate and won't even start a span. Does that cover your use case or does it need to be more complex?

**Reply by chad-g-adams** [reply]
Interesting! I'm looking for something almost the reverse of that - to boost/increase the sampling rate (or even force the trace to be always sampled) based on the presence of a header

**Reply by bryannaegele** [reply]
Ok. We just need to implement https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-and-response-headers which would unlock that
